File: src/typescript/src/pipeline_services/examples/example04_json_loader_remote_get.ts
import { PipelineJsonLoader, PipelineRegistry } from "../../index.js";

async function main(): Promise<void> {
  const jsonText = `
{
  "pipeline": "example04_json_loader_remote_get",
  "type": "unary",
  "actions": [
    {
      "name": "remote_get_fixture",
      "$remote": {
        "endpoint": "http://127.0.0.1:8765/remote_hello.txt",
        "method": "GET",
        "timeoutMillis": 1000,
        "retries": 0
      }
    }
  ]
}
`;

  const pipeline = new PipelineJsonLoader().load_str(
    jsonText,
    new PipelineRegistry(),
  );
  const output = await pipeline.run("ignored");
  // eslint-disable-next-line no-console
  console.log(String(output));
}

void main();

File: src/Python/pipeline_services/examples/example04_json_loader_remote_get.py
from __future__ import annotations

from pipeline_services.config.json_loader import PipelineJsonLoader
from pipeline_services.core.registry import PipelineRegistry


def main() -> None:
    json_text = """
{
  "pipeline": "example04_json_loader_remote_get",
  "type": "unary",
  "steps": [
    {
      "name": "remote_get_fixture",
      "$remote": {
        "endpoint": "http://127.0.0.1:8765/remote_hello.txt",
        "method": "GET",
        "timeoutMillis": 1000,
        "retries": 0
      }
    }
  ]
}
"""

    registry = PipelineRegistry()
    loader = PipelineJsonLoader()
    pipeline = loader.load_str(json_text, registry)
    result = pipeline.run("ignored")
    print(result.context)


if __name__ == "__main__":
    main()
